packages/protocol/socket.ts

```diff
@@ -252,11 +253,14 @@ const doSend = makeInfallible(message => {
   gSentBytes += stringified.length;
 
   gSessionCallbacks?.onRequest(message);
-
   socket.send(stringified);
 });
 
 function flushQueuedMessages() {
+  if (socket.readyState != 1) {
+    return;
+  }
```

Comment:
These two checks are needed to prevent HMR issues. I think they're fairly safe to land because we only want to send messages when the socket is in fact connected